src/leaguepedia_client.py

```
@@ -93,31 +148,332 @@ async def fetch_upcoming_matches(self, tournament_name_like: str):
+    ) -> None:
+        """
+        Handle a rate-limit exception by applying backoff and notifying.
+
+        Pauses briefly to avoid retry storms, applies exponential backoff,
+        and schedules an admin notification.
+        """
+        # Brief pause to avoid immediate retry storms from concurrent tasks
+        try:
+            await asyncio.sleep(5)
+        except asyncio.CancelledError:
+            raise
```

Ruff (0.14.10)

Remove exception handler; error is immediately re-raised

(TRY203)